src/grammar.coffee

```diff
@@ -166,20 +166,41 @@ grammar =
   ]
 
   String: [
-    o 'STRING',                                 -> new StringLiteral $1
-    o 'STRING_START Body STRING_END',           -> new StringWithInterpolations $2
+    o 'STRING', ->
+      new StringLiteral(
+        $1.slice 1, -1 # strip artificial quotes and unwrap to primitive string
+        quote:        $1.quote
+        initialChunk: $1.initialChunk
+        finalChunk:   $1.finalChunk
+        indent:       $1.indent
+        double:       $1.double
+        heregex:      $1.heregex
+      )
+    o 'STRING_START Interpolations STRING_END', -> new StringWithInterpolations Block.wrap($2), quote: $1.quote
+  ]
+
+  Interpolations: [
+    o 'InterpolationChunk',                     -> [$1]
+    o 'Interpolations InterpolationChunk',      -> $1.concat $2
+  ]
+
+  InterpolationChunk: [
+    o 'INTERPOLATION_START Body INTERPOLATION_END',                -> new Interpolation $2
+    o 'INTERPOLATION_START INDENT Body OUTDENT INTERPOLATION_END', -> new Interpolation $3
+    o 'INTERPOLATION_START INTERPOLATION_END',                     -> new Interpolation
+    o 'String',                                                    -> $1
   ]
 
   Regex: [
-    o 'REGEX',                                  -> new RegexLiteral $1
-    o 'REGEX_START Invocation REGEX_END',       -> new RegexWithInterpolations $2.args
+    o 'REGEX',                                  -> new RegexLiteral "#{$1}", delimiter: $1.delimiter
```

Comment:
I don’t see `toPrimitiveString` yet, maybe you haven’t pushed that change? But regardless, I agree that `"#{$1}"` is slick, and it would be a shame not to use it. I think `toPrimitiveString` doesn’t explain enough, i.e. it’s not clear why we need to convert anything into a primitive string. A comment like ``# Convert `String` object back to primitive string now that we’ve retrieved stowaway extra properties.`` is clearer, though it begs the question how often it should be repeated down the file. Alternatively I guess you could explain it in a comment inside `toPrimitiveString`.